public/js/reader.js

```diff
@@ -1,408 +1,601 @@
+    }
+
+    Reader.applyContainerWidth();
+
+    Reader.currentPageLoaded = false;
+    // display overlay if it takes too long to load a page
+    setTimeout(() => {
+        if (!Reader.currentPageLoaded) { $("#i3").addClass("loading"); }
+    }, 500);
+
+    // update full image link
+    $("#imgLink").attr("href", Reader.pages[Reader.currentPage]);
+
+    // Send an API request to update progress on the server
+    genericAPICall(`api/archives/${Reader.id}/progress/${Reader.currentPage + 1}`, "PUT", null, "Error updating reading progress!", null);
+
+    // scroll to top
+    window.scrollTo(0, 0);
+
+    // Update url to contain all search parameters, and push it to the history
+    if (!fromHistory) { window.history.pushState(null, null, `?id=${Reader.id}&p=${page + 1}`); }
+};
+
+Reader.toggleFitMode = function (e) {
+    // possible options: fit-container, fit-width, fit-height
+    Reader.fitMode = localStorage.fitMode = e.target.id;
+    $("#fit-mode input").removeClass("toggled");
+    $(e.target).addClass("toggled");
+
+    if (Reader.fitMode === "fit-container") {
+        $("#container-width").show();
+    } else {
+        $("#container-width").hide();
+    }
+    Reader.applyContainerWidth();
+};
+
+Reader.registerContainerWidth = function () {
+    // Examples of allowed values: 1200, 1200px, 90%
+    // Default value: 1200px
+    const raw = $("#container-width-input").val().trim();
+    if (!raw) { // fall back to default
+        delete Reader.containerWidth;
+        localStorage.removeItem("containerWidth");
+    } else {
+        let value, type;
+
+        [, value, type] = /^(\d+)(px|%)?$/.exec(raw);
+        value ||= 1200;
+        type ||= "px";
+
+        Reader.containerWidth = localStorage.containerWidth = `${value}${type}`;
+    }
+    Reader.applyContainerWidth();
+};
+
+Reader.applyContainerWidth = function () {
+    $(".reader-image, .sni").attr("style", "");
+
+    if (Reader.fitMode === "fit-height") {
+        // Fit to height forces the image to 90% of visible screen height.
+        // If the header is hidden, or if we're in infinite scrolling, then the image
+        // can take up to 98% of visible screen height because there's more free space
+        $(".reader-image").attr("style", `max-height: ${Reader.hideHeader || Reader.infiniteScroll ? 98 : 90}vh;`);
+    } else if (Reader.fitMode === "fit-width") {
+        $(".reader-image").attr("style", "width: 100%;");
+        $(".sni").attr("style", "max-width: 98%");
+    } else if (Reader.containerWidth) {
+        // If the user defined a custom width, then we can fall back to that one
+        $(".sni").attr("style", `max-width: ${Reader.containerWidth}`);
+        $(".reader-image").attr("style", "width: 100%");
+    } else if (!Reader.showingSinglePage) {
+        // Otherwise, if we are showing two pages we can override the default width
+        $(".sni").attr("style", "max-width: 90%");
```

Comment:
I removed the max-width from lrr.css and added width: fit-content to .sni in that section. That should fix the container issue too.

Reply:
`fit-content` is chrome-only for now it seems, you also need to add `moz-fit-content` for it to work on FF.

Reply:
Damnit mozilla. Fixed. 

This looks more and more like it should go into a CSS file. It would be more elegant to just toggle a class or ad-hoc property in the body and let CSS take it from there, rather than clearing and replacing the style every time, but we'll leave that for a larger CSS rework I guess. Don't want to alter the PR too much or it'll also be more work for you.